src/Proto3/Suite/Class.hs

```diff
@@ -405,16 +407,16 @@ instance Primitive BL.ByteString where
   primType _ = Bytes
 
 instance forall e. (Bounded e, Named e, Enum e) => Primitive (Enumerated e) where
-  encodePrimitive num = Encode.enum num . enumify . enumerated
+  encodePrimitive num = Encode.enum num . enumify . coerce @(Enumerated e) @(Either Int e)
     where enumify (Left i) = i
           enumify (Right x) = fromEnum x
-  decodePrimitive = fmap Enumerated Decode.enum
-  primType _ = Named (Single (nameOf (Proxy @e)))
+  decodePrimitive = coerce @(_ (Either Int e)) @(_ (Enumerated e)) Decode.enum
+  primType _ = Named (Single (nameOf (proxy# :: Proxy# e)))
 
 instance (Primitive a) => Primitive (ForceEmit a) where
-  encodePrimitive num = encodePrimitive num . forceEmit
-  decodePrimitive     = fmap ForceEmit decodePrimitive
-  primType _          = primType (Proxy @a)
+  encodePrimitive num = encodePrimitive num . coerce @(ForceEmit a) @a
```

Comment:
As mentioned above, the original code is probably just as efficient as the new code.  I find the original code clearer.  If there is any benefit from using `coerce` in this case, which I doubt, then it would only arise if you coerced the entire function, not merely its argument.

Reply:
reverted in 9309ab1